src/main/java/com/github/manolo8/darkbot/modules/utils/NpcAttacker.java

```diff
@@ -143,9 +145,17 @@ private boolean shouldSab() {
     }
 
     private boolean shouldRsb() {
-        if (!main.config.LOOT.RSB.ENABLED || !target.npcInfo.extra.has(NpcExtra.USE_RSB)) return false;
-        if (usedRsb < System.currentTimeMillis() - main.config.LOOT.RSB.AMMO_REFRESH) usedRsb = System.currentTimeMillis();
-
+        if (!main.config.LOOT.RSB.ENABLED || main.config.LOOT.RSB.KEY == null
+                || !target.npcInfo.extra.has(NpcExtra.USE_RSB)) return false;
+        String itemId = main.facadeManager.slotBars.standardBar.slots.stream()
```

Comment:
Previous approach which i abandoned because couldn't construct main was having it ready in SlotbarsProxy. Future-ready with ordinal check. https://github.com/ODAKAB/DarkBot/commit/da4319845ac670015feb93a51811d5e02b731f2f